modules/openxr/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,190 @@
+	HashMap<String, bool *> request_extensions;
+
+	request_extensions[XR_FB_SPACE_WARP_EXTENSION_NAME] = &available;
+
+	return request_extensions;
+}
+
+void *OpenXRFbSpaceWarpExtensionWrapper::set_system_properties_and_get_next_pointer(void *p_next_pointer) {
+	return &system_space_warp_properties;
+}
+
+void *OpenXRFbSpaceWarpExtensionWrapper::set_projection_views_and_get_next_pointer(int p_view_index, void *p_next_pointer) {
+	if (enabled) {
+		return &space_warp_info[p_view_index];
+	} else {
+		return nullptr;
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::on_session_created(const XrSession p_instance) {
+	OpenXRAPI *openxr_api = OpenXRAPI::get_singleton();
+	if (!openxr_api) {
+		return;
+	}
+
+	openxr_api->register_projection_views_extension(this);
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::on_session_destroyed() {
+	OpenXRAPI *openxr_api = OpenXRAPI::get_singleton();
+	if (!openxr_api) {
+		return;
+	}
+
+	openxr_api->unregister_projection_views_extension(this);
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::on_state_ready() {
+	OpenXRAPI *openxr_api = OpenXRAPI::get_singleton();
+
+	if (!openxr_api) {
+		return;
+	}
+
+	int width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+	int view_count = openxr_api->get_xr_interface()->get_view_count();
+
+	motion_vector_swapchain_info.create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT, GL_RGBA16F, width, height, 1, view_count);
+	motion_vector_depth_swapchain_info.create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT, GL_DEPTH24_STENCIL8, width, height, 1, view_count);
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::on_main_swapchains_created() {
+	OpenXRAPI *openxr_api = OpenXRAPI::get_singleton();
+	int view_count = openxr_api->get_xr_interface()->get_view_count();
+
+	XrCompositionLayerProjectionView *projection_views = openxr_api->get_projection_views();
+	XrPosef identity_pose = { { 0.0, 0.0, 0.0, 1.0 }, { 0.0, 0.0, 0.0 } };
+
+	space_warp_info = memnew_arr(XrCompositionLayerSpaceWarpInfoFB, 2);
+	for (int i = 0; i < view_count; i++) {
+		space_warp_info[i].type = XR_TYPE_COMPOSITION_LAYER_SPACE_WARP_INFO_FB;
+
+		space_warp_info[i].next = nullptr;
+
+		space_warp_info[i].layerFlags = 0;
+
+		space_warp_info[i].motionVectorSubImage.swapchain = motion_vector_swapchain_info.get_swapchain();
+		space_warp_info[i].motionVectorSubImage.imageRect.offset.x = 0;
+		space_warp_info[i].motionVectorSubImage.imageRect.offset.y = 0;
+		space_warp_info[i].motionVectorSubImage.imageRect.extent.width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+		space_warp_info[i].motionVectorSubImage.imageRect.extent.height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+		space_warp_info[i].motionVectorSubImage.imageArrayIndex = i;
+
+		space_warp_info[i].appSpaceDeltaPose = identity_pose;
```

Comment:
Great work and and a much needed feature for XR development.

For this appSpaceDeltaPose parameter, [OpenXR specs](https://registry.khronos.org/OpenXR/specs/1.1/html/xrspec.html#XR_FB_space_warp) and [Meta documentation](https://developers.meta.com/horizon/documentation/native/android/mobile-asw/) recommend filling it with _Inv(appSpacePoseInPrevFrame) * appSpacePoseInCurrFrame_ instead of identity matrix to take into account artificial locomotion. Would it be something we may want to compute each frame or only when there is a change in pose?

Reply:
The bits of this PR integrating the `XR_FB_space_warp` extension are just for testing the motion vectors at this point. Eventually, the OpenXR bits will be moved into the 'godot_openxr_vendors' extension, and the plan is to finish out that implementation at that point.

But, yes, any time the `XROrigin` is moved, we'll need to update the `appSpaceDeltaPose` field on that struct.